homeassistant/components/geonetnz_volcano/config_flow.py

```diff
@@ -57,7 +58,7 @@ async def async_step_user(self, user_input=None):
         if identifier in configured_instances(self.hass):
             return await self._show_form({"base": "already_configured"})
 
-        if self.hass.config.units.name == CONF_UNIT_SYSTEM_IMPERIAL:
+        if self.hass.config.units.length_unit == LENGTH_MILES:
```

Comment:
Checking/Storing the `name` of the unit system will stop working if we rename or add a new unit system.
It is more future-proof to check the `length_unit` of the unit system, even if the component still uses "imperial" terminology.

Reply:
Adjusted to use `hass.config.units == ***`